htsinfer/rand_nuc_seq.py

```diff
@@ -0,0 +1,249 @@
+from collections import Counter
+
+import numpy as np
+
+def randomize_nucleotide_sequence(input_sequences: list,
+    number_random_seq: int = 1, min_prob: float = 0) -> list:
+    """
+    Returns randomised sequence adhering to the dinucleotide probabilites.
+
+    Args:
+        input_sequences: List of input sequences as strings.
+        number_random_seq: Number of randomised sequences returned per input
+            sequence.
+        min_prob: Minimal probability of dinucleotide to appear. If set to -1
```

Comment:
It's a really nice idea to include this, but I don't find it very intuitive that setting it to 0 will actually mean that 0 is _not_ possible. I would switch around what 0 and -1 mean here...

Reply:
I agree. Actually, Thomas and I thought to leave the min_prob out for now. Apparently, we forgot to clean up the docstrings properly, as we were planning to implement it at the beginning. Our current solution is that, when the dinucleotide occurrences are counted to calculate the markov matrix, we start counting at 1 (so we assume that each combination has already occurred once). Thus, if a certain combination never occurs in the input, it still gets a probability. For short input sequences this is not the most elegant solution and leads to incorrect probabilities, but the problem should be negligible if the input sequences are of sufficient length. We did not do it with the min_prob as we were a bit short on time and wanted to submit something before tomorrow, so maybe we can give it a few more minutes and try to implement it properly.

Reply:
Your solution includes a 'pseudo-count' of 1, which is fine. As you hint in the above explanation, the pseudo-count value sets the scale for how much data you need to see before you start trusting the counts. If you want to implement an additional parameter, it should be this pseudo-count.